### PR TITLE
feat(schema): [sc 2922] HTTP request validation using openapi v3

### PR DESCRIPTION
**Pull request by mahad-akhter**
- Update Schema KV Store name to match other KV store names convention
- Add Schema Validation function in filter function
- Update storage to Open API spec file so that each operation is stored separately in the KV store with key being in the format:
"<service_id>-<operation_id>"
- While adding the Reference Schema to the KV store, the optional fields in the Request Body are set to nullable=True manually

### CODE REVIEW

File: nozl/schema/schema.go [hunk 1]
```diff
@@ -30,42 +35,122 @@ func ParseOpenApiV3Schema(serviceID string, specFile []byte) error {
...
-			schemaList[pathValue.Post.OperationID] = addSchema(pathKey, "post", pathValue.Post)
-		} else if pathValue.Put != nil {
-			schemaList[pathValue.Put.OperationID] = addSchema(pathKey, "put", pathValue.Put)
-		} else if pathValue.Patch != nil {
-			schemaList[pathValue.Patch.OperationID] = addSchema(pathKey, "patch", pathValue.Patch)
-		} else if pathValue.Delete != nil {
-			schemaList[pathValue.Delete.OperationID] = addSchema(pathKey, "delete", pathValue.Delete)
+func MakeOptionalFieldsNullable(operation *openapi3.Operation) error {
+	if operation.RequestBody != nil {
+		ReqBodySchema := operation.RequestBody.Value.Content["application/x-www-form-urlencoded"].Schema.Value
```
Comment (taimoor-mirza) [inline nozl/schema/schema.go:63]: @mahad-akhter This header may break in other OAPI specs. Add a warning here at least.

File: nozl/schema/schema.go [hunk 2]
```diff
@@ -30,42 +35,122 @@ func ParseOpenApiV3Schema(serviceID string, specFile []byte) error {
...
 
-	return nil
+	msgBody := msg.Body
+	fmt.Println(schemaValid)
+	fmt.Println(msgBody)
+
+	ctx := context.Background()
+	jsonData, _ := json.Marshal(&msgBody)
+	formData := strings.NewReader(string(jsonData))
+	//baseUrl := "https://api.twilio.com"
```
Comment (taimoor-mirza) [inline nozl/schema/schema.go]: @mahad-akhter Remove commented out lines if they aren't required.

File: nozl/schema/schema.go [hunk 3]
```diff
@@ -30,42 +35,122 @@ func ParseOpenApiV3Schema(serviceID string, specFile []byte) error {
...
+	//rgx, _ := regexp.Compile("{AccountSid}")
+	//ep := rgx.ReplaceAllString(schemaValid.Path, "AC9f560ea30baaaf8013e4e44284eb6768")
+	//data := url.Values{}
+
+	//data.Add("To", +923244253153)
+	//data.Add("Body", "helloworld")
+	//payload := strings.NewReader("Body=Hello%20World!&To=%2B923214930471&From=%2B19034598701&edfgsgd=sdfds")
+	//payload := strings.NewReader(data.Encode())
+
+	httpReq, err := http.NewRequest("POST", schemaValid.Path, formData)
```
Comment (taimoor-mirza) [inline nozl/schema/schema.go]: Replace hard-coded HTTP method with the one specified in the OAPI spec

File: nozl/schema/schema.go (hunk 1 above)
Comment (mahad-akhter) [reply nozl/schema/schema.go:63]: By warning, you mean a ToDo to take care of this later?

Comment (mahad-akhter) [reply nozl/schema/schema.go:63]: Done

File: nozl/schema/schema.go (hunk 2 above)
Comment (mahad-akhter) [reply nozl/schema/schema.go]: Done

File: nozl/schema/schema.go (hunk 3 above)
Comment (mahad-akhter) [reply nozl/schema/schema.go]: Done

**Review by taimoor-mirza — APPROVED** [review]

**Pull request merged**